### PR TITLE
Events not being added to outlook

### LINKED ISSUE #7435: [CAL-1161] Events not being added to Outlook calendar
If the email associated with [cal.com](http://cal.com) account isn't outlook, destination calendar for an event type set to outlook doesn't add it.

Created via Threads. See full discussion: [https://threads.com/34452351117](https://threads.com/34452351117)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1161](https://linear.app/calcom/issue/CAL-1161/events-not-being-added-to-outlook-calendar)</sub>

### PR DESCRIPTION
**Pull request by keithwillcode**
## What does this PR do?

Fixes #7435 

Events were not created in Outlook if a specific external calendar ID wasn't set.

**Environment**: Staging(main branch) / Production

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [x] Create new events that are destined for Outlook
- Reschedule existing events that are destined for Outlook (this is broken but will require another larger PR to fix. It's not specific to outlook but seemingly to all calendars based on how the logic for references is setup in the booking hander

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: packages/app-store/office365calendar/lib/CalendarService.ts [hunk 1]
```diff
@@ -55,11 +55,11 @@ export default class Office365CalendarService implements Calendar {
 
   async createEvent(event: CalendarEvent): Promise<NewCalendarEventType> {
     try {
-      const calendarId = event.destinationCalendar?.externalId
-        ? `${event.destinationCalendar.externalId}/`
-        : "";
+      const eventsUrl = event.destinationCalendar?.externalId
```
Comment (keithwillcode) [inline packages/app-store/office365calendar/lib/CalendarService.ts:58]: See [docs](https://learn.microsoft.com/en-us/graph/api/calendar-post-events?view=graph-rest-1.0&tabs=http).

We were generating a URL of `/me/calendars/events` when there's no externalId but the URL needs to be `/me/calendar/events` in that case. Upon debugging, I saw that Outlook 365 was sending us back a 405.

Screen Shot 2023-03-14 at 10 53 03 AM

Comment (joeauyeung) [reply packages/app-store/office365calendar/lib/CalendarService.ts:58]: I'm surprised this is showing up now. How did you manage to get the 405 response?

Comment (keithwillcode) [reply packages/app-store/office365calendar/lib/CalendarService.ts:58]: I manually added a console log to see the response. I'll probably spend some more time to see why if these errors aren't being logged currently.

**Review by joeauyeung — APPROVED** [review]
Tested with normal user flow and by deleting the `externalId` from the destination calendar. All working as expected. Thank you @keithwillcode

**Pull request merged**